### DISCUSSION TITLE
Meeting Minutes from January 11, 2024

### DISCUSSION THREAD

**Discussion by dezidizon** [CVWRSM/cvwrsm#37] [Meeting Minutes]
**Open Action Items:**

- Yiwei Cheng will provide historical study to use for comparing results from existing WRIMSv2.jar and updated WRIMSv2.jar. This is tracked as #36.
- Provide access to Hao Xie to review the updates for the WRIMS DevOps work. 
- Coordinate with Hao Xie for manual steps in creating WRIMS installer and patch artifacts. This is tracked as #29 
- Apply copyright files (license and source) in the wrims repository using Eclipse Public License (EPL) version 2 or 3. This is tracked as #16.
- RMA has sent a DSS global variable flag for Hao Xie to use in order to force WRIMS to write DSS6 files. RMA will help diagnosing the issues as this is not working correctly. This is tracked as #31.
- Remove vista libraries from WRIMS. This is tracked as #33 .
- Document descriptions for each folder in WRIMS. This is tracked as #36.

**Discussions:**
- The wrims2 repository was moved to the CentralValleyModeling organization from CVWRSM and was renamed to wrims. 
    - Permissions for the CentralValleyModeling organization were temporarily adjusted to "owner" for Deziree Dizon to make adjustments to the WRIMS DevOps project and to add Tom Evans and Adam Korynta as users.
- Adam Korynta was provided the changelist to help Hao Xie address issues with writing DSS6 files. This is related to #31.
- Tom Evans discussed work completed for the WRIMS DevOps refactors.
    - Tom is using IntelliJ to compile WRIMS. Any IDE may be used to compile WRIMS.
    - The structure of directories have been reorganized to fit the gradle build file. the directory structure is maintained and the content of the source files are adjusted at minimum and as necessary to have the build work.
    - Work started with the WRIMSv2.jar because this does not depend on other java files that are built.
    - The new folder of wrims-core was created which includes classes identical to those contained in the wrimsv2.jar. 
    - This folder is labeled as wrims-core following a common term used in HEC development to signify that it is a core module library or artifact that many are dependent on. A change here affects many things.
- Binary files will be moved and presented as artifacts instead. With the new arrangement, the jars are pulled in at build time from a remote maven repository with a specified version needed for the build.  
    - Peter added that many software projects source artifacts within maven central to be able to do builds. Once the artifact is into the gradle build and a local repository is established in a machine, these will then be written into a local file. 
    - Binary dependencies that are not available or 3rd party services have been put into a local dependencies folder.
- Hao will review the work done in the fork for wrims devops. 
    - The intention is to work closely with Hao to make sure the the dependency of the code is honored and Hao is aware and approves the updates.
    - Naming of the packages will be discussed.
- Change history for the repository has been preserved in updating the directory structure of WRIMS. 
    - Once work is done for restructuring and the new build of WRIMS has been accepted, history can be pruned.
    - History is kept in the .git folder when the repository is cloned.
    - Adam noted that performing certain operations such as going back in history will increase the size of the history folder. The aim is to keep the size of the folder as light as possible to keep operations smoother.
- For testing the updated WRIMS 2 jar, this will be dropped into its corresponding location in the distributed WRIMS 2 package to replace the WRIMS_v2.jar. The results must be the same as running the current WRIMS_v2.jar.
    - Yiwei Cheng will send a historical study to use for testing.
    - The WRIMSv2_Engine.bat may be run to reference a study's _study.config file. The study will have to be run using the UI to generate the _study.config file. After this is generated, the batch file may be run.
    - Tom will consul Stephen Ackerman regarding comparing model results in DSS files.
- Naming for modules will be addressed next.

**Comment by adamkorynta** [comment]
Additional information and decisions need to be made regarding jar and DLL files that are not available in public maven repositories. Some are licensed software that would not be allowed to be publicly distributed.

Tom will follow-up with Hao with a listing of these libraries in order to decide how they are added to an artifact management system.

**Comment by rma-tom** [comment]
Links to issues cited don't match visible URLs. Visible text for issue 29 "https://github.com/CentralValleyModeling/wrims/issues/29\" points to https://github.com/CentralValleyModeling/wrims/discussions/url

**Reply by dezidizon** [reply]
@rma-tom Thank you for catching that! Updated the references to the issues in the discussions.

**Comment by rma-psmorris** [comment]
@hxiedwr When you manually check the schematic perspective for testing purposes, what do you look for?

**Comment by hxiedwr** [comment]
I do the follows things:


  1.  open schematic xml file under the “schematic” folder.
  2.  zoom to a small region
  3.  load dv and sv dss files in the dss perspective
  4.  compare data on the schematic with selecting time period/step on the drop down list

Thank you,

Hao

  1.

________________________________
